libmamba/src/specs/version.cpp

```diff
@@ -0,0 +1,562 @@
+                    return -1;
+                }
+                if (l == "")
+                {
+                    return 1;
+                }
+                if (l == "post")
+                {
+                    return 2;
+                }
+                return 0;
+            };
+            const auto a_lit_val = lit_priority(a.litteral());
+            const auto b_lit_val = lit_priority(b.litteral());
+            // If two regular string, we need to use string comparison
+            if ((a_lit_val == 0) && (b_lit_val == 0))
+            {
+                return compare_three_way<std::string>(a.litteral(), b.litteral());
+            }
+            return compare_three_way(a_lit_val, b_lit_val);
+        }
+    }
+
+    auto VersionPartAtom::operator==(const VersionPartAtom& other) const -> bool
+    {
+        // More efficient thatn three way comparison because of edge cases
+        auto attrs = [](const VersionPartAtom& a) -> std::tuple<std::size_t, const std::string&> {
+            return { a.numeral(), a.litteral() };
+        };
+        return attrs(*this) == attrs(other);
+    }
+
+    auto VersionPartAtom::operator!=(const VersionPartAtom& other) const -> bool
+    {
+        // More efficient thatn three way comparison
+        return !(*this == other);
+    }
+
+    auto VersionPartAtom::operator<(const VersionPartAtom& other) const -> bool
+    {
+        return compare_three_way(*this, other) == strong_ordering::less;
+    }
+
+    auto VersionPartAtom::operator<=(const VersionPartAtom& other) const -> bool
+    {
+        return compare_three_way(*this, other) != strong_ordering::greater;
+    }
+
+    auto VersionPartAtom::operator>(const VersionPartAtom& other) const -> bool
+    {
+        return compare_three_way(*this, other) == strong_ordering::greater;
+    }
+
+    auto VersionPartAtom::operator>=(const VersionPartAtom& other) const -> bool
+    {
+        return compare_three_way(*this, other) != strong_ordering::less;
+    }
+
+    /*******************************
+     *  Implementation of Version  *
+     *******************************/
+
+    Version::Version(std::size_t epoch, CommonVersion&& version, CommonVersion&& local) noexcept
+        : m_version{ std::move(version) }
+        , m_local{ std::move(local) }
+        , m_epoch{ epoch }
+    {
+    }
+
+    auto Version::epoch() const noexcept -> std::size_t
+    {
+        return m_epoch;
+    }
+
+    auto Version::version() const noexcept -> const CommonVersion&
+    {
+        return m_version;
+    }
+
+    auto Version::local() const noexcept -> const CommonVersion&
+    {
+        return m_local;
+    }
+
+    auto Version::str() const -> std::string
+    {
+        return fmt::format("{}", *this);
+    }
+
+    namespace
+    {
+        /**
+         * Compare two range where some trailing elements can be considered as empty.
+         *
+         * If ``0`` is considered "empty" then all the ranges ``[1, 2] and ``[1, 2, 0]``,
```

Comment:
What is this double anti-quote syntax?

Reply:
Same as `0` in makrdown but also works with reStructured (Sphinx).

Reply:
Ah, I thought we were using Markdown over Doxygen syntax